### PR TITLE
grate: consolidate cull-flags

### PR DESCRIPTION
**Pull request by kusma**
This allows us to use OpenGL semantics, and reduces the amount of
function calls a tad.

Depends on grate-driver/envytools#1

### CODE REVIEW

File: src/libgrate/grate-3d.c [hunk 1]
```diff
@@ -145,6 +145,25 @@ static void grate_3d_set_guardband(struct host1x_pushbuf *pb,
...
+		return ctx->tri_face_front_cw ? CULL_FACE_CW : CULL_FACE_CCW;
+
+	case GRATE_CULL_FACE_BACK:
+		return ctx->tri_face_front_cw ? CULL_FACE_CCW : CULL_FACE_CW;
+
+	case GRATE_CULL_FACE_BOTH:
+		return CULL_FACE_BOTH;
+	}
+
+	return CULL_FACE_NONE; /* just a stupid fallback */
```
Comment (digetx) [inline src/libgrate/grate-3d.c:164]: I'd print a error message and abort on unexpected "default:", that will avoid stupidity.

As I said before, it should be better to move all context params checks/mappings to the grate-3d-ctx.c, but that should be another patch. I think the current variant is good enough.

**Comment by digetx** [comment]
Great, thank you! I already merged envytools.

File: src/libgrate/grate-3d.c (hunk 1 above)
Comment (kusma) [reply src/libgrate/grate-3d.c:164]: Yeah, that was my reasoning as well; this move can come later 👍

Comment (kusma) [reply src/libgrate/grate-3d.c:164]: And yeah, I guess it would make sense to print (or just assert) in default instead.
